blockchain/vm/instructions.go

```diff
@@ -884,11 +884,11 @@ func opStop(pc *uint64, evm *EVM, scope *ScopeContext) ([]byte, error) {
 	return nil, nil
 }
 
-func opSuicide(pc *uint64, evm *EVM, scope *ScopeContext) ([]byte, error) {
+func opSelfdestruct(pc *uint64, evm *EVM, scope *ScopeContext) ([]byte, error) {
 	balance := evm.StateDB.GetBalance(scope.Contract.Address())
 	evm.StateDB.AddBalance(common.BigToAddress(scope.Stack.pop()), balance)
 
-	evm.StateDB.Suicide(scope.Contract.Address())
+	evm.StateDB.SelfDestruct(scope.Contract.Address())
```

Comment:
https://github.com/ethereum/go-ethereum/commit/988d84aa7caf8e71ce441fa65f80d44216d9e00e#diff-7da47ad076053fe0f266b3088eb9ee206c7f3f8fe5a93116a3f5e25ee5074213R829

Previous SelfDestruct halts as well. Could there be a EIP that we missed?

Reply:
I think we should add this logic as well:
https://github.com/ethereum/go-ethereum/commit/1fa91729f2a591df2baf2dc77e81711a6e61c028#diff-121e1c875cf3bed2f44305ab103a23126e330633fc45011cb5b1041f7b1ca119R269-R271

While we're adding `errStopToken`, should we make other functions to return it as well? (like `opReturn`)